### PR TITLE
Add ECMWF consumer task

### PR DESCRIPTION
**Pull request by devsjc**
:copilot summary

This pull requests changes the way secrets are mapped into the consumer container - so the secrets will need to have their KEYS updated for the nwp and nwp-national services before appy

### CODE REVIEW

File: terraform/modules/services/nwp/iam.tf [hunk 1]
```diff
@@ -38,7 +38,7 @@ resource "aws_iam_policy" "nwp-secret-read" {
           "secretsmanager:GetSecretValue",
         ]
         Effect   = "Allow"
-        Resource = data.aws_secretsmanager_secret_version.nwp-api-version.arn
+        Resource = data.aws_secretsmanager_secret.nwp-consumer-secret.arn
```
Comment (peterdudfield) [inline terraform/modules/services/nwp/iam.tf:41]: how come the `version` has been dropped here?

File: terraform/modules/services/nwp/secrets.tf [hunk 2]
```diff
@@ -1,11 +1,7 @@
...
-data "aws_secretsmanager_secret" "nwp-api" {
+data "aws_secretsmanager_secret" "nwp-consumer-secret" {
   name = "${var.environment}/consumer/${var.consumer-name}"
   #  arn = "arn:aws:secretsmanager:eu-west-2::secret:development/consumer/nwp"
 }
 
-data "aws_secretsmanager_secret_version" "nwp-api-version" {
-  secret_id = data.aws_secretsmanager_secret.nwp-api.id
-  #  arn = "arn:aws:secretsmanager:eu-west-2::secret:development/consumer/nwp"
-}
```
Comment (peterdudfield) [inline terraform/modules/services/nwp/secrets.tf:11]: ah, perhaps this isnt need any more?

**Review by peterdudfield — APPROVED** [review]
Ive just put in one question, but it looks really good.

File: terraform/modules/services/nwp/secrets.tf (hunk 2 above)
Comment (devsjc) [reply terraform/modules/services/nwp/secrets.tf:11]: It still is needed, I got confused about how to access the arn in the secret!

File: terraform/modules/services/nwp/iam.tf (hunk 1 above)
Comment (devsjc) [reply terraform/modules/services/nwp/iam.tf:41]: By mistake!

**Pull request merged**